### ISSUE TITLE
Rounding error for np.complex128 array multiplication from numpy v1.19.0

### ISSUE THREAD

**Issue by antalszava** [numpy/numpy#17602]
When using `numpy<=1.18.4`, an example `np.complex128` array multiplied with itself returns the identity (up to precision) as expected. With `numpy>=1.19.0`, however, the complex part of certain elements are still there (and lead to an eventual deviation in computations).

Perhaps related in some way to [`Deprecation of round for np.complexfloating scalars`](https://numpy.org/devdocs/release/1.19.0-notes.html#deprecation-of-round-for-np-complexfloating-scalars).

### Reproducing code example:

```python
import numpy as np

m = 1
n = 2
BS = np.identity(4, dtype=np.complex128)
BS[m, m] = 1.0 / np.sqrt(2)
BS[m, n] = 1.0j / np.sqrt(2)
BS[n, m] = 1.0j / np.sqrt(2)
BS[n, n] = 1.0 / np.sqrt(2)
np.set_printoptions(precision=25)
print(BS @ BS)
```

### Error message:

The code example outputs (with version information):

*1.18.4 3.7.6 (default, Jan  8 2020, 19:59:22) [GCC 7.3.0]*
```
[[1.+0.j                 0.+0.j                 0.+0.j
  0.+0.j                ]
 [0.+0.j                 0.+0.j                 0.+0.9999999999999998j
  0.+0.j                ]
 [0.+0.j                 0.+0.9999999999999998j 0.+0.j
  0.+0.j                ]
 [0.+0.j                 0.+0.j                 0.+0.j
  1.+0.j                ]]
```
*1.19.2 3.8.1 (default, Jan  8 2020, 22:29:32) [GCC 7.3.0]*
```
[[ 1.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.j                ]
 [ 0.0000000000000000e+00+0.j
  -2.2371143170757382e-17+0.j
   0.0000000000000000e+00+0.9999999999999998j
   0.0000000000000000e+00+0.j                ]
 [ 0.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.9999999999999998j
   2.2371143170757382e-17+0.j
   0.0000000000000000e+00+0.j                ]
 [ 0.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.j
   0.0000000000000000e+00+0.j
   1.0000000000000000e+00+0.j                ]]
```

### NumPy/Python version information:

Included in the previous section.

**Comment by seberg** [comment]
since this uses `@`, there is a good chance that this is actually caused by a difference in OpenBLAS. Similar changes in numerical results may also happen if you are changing operating system or hardware.

The result in both cases is inexact giving `+0.9999999999999998j`, in one case, so I expect this is the normal precision guaranteed by floating point math on the CPU?

**Comment by josh146** [comment]
> since this uses @, there is a good chance that this is actually caused by a difference in OpenBLAS. 

Ah, so could this be due to a change in the OpenBLAS version packaged in the NumPy `manylinux` wheel?

**Comment by seberg** [comment]
Yes, the matrix multiplication will be performed by the BLAS system here.

**Comment by charris** [comment]
@seberg I checked with ATLAS and it is still there and also for `dot`.
